### PR TITLE
Use adaptive SVG favicon for rustdoc like other rust sites

### LINKED ISSUE #72165: Add a white glow or background on the favicon.
Hi,

The Rust favicon is black and transparent. On dark browser themes, it is not easy to see. Since rust book and github also have this particularity, it's hard to find the tab among the other ones.

Not really important, but if it is cheap to fix that would help :)

My tabs right now:
image

### PR DESCRIPTION
**Pull request by Cldfire**
Use the theme-adaptive SVG favicon that was recently introduced [for the Rust site](https://github.com/rust-lang/www.rust-lang.org/pull/1185) (and others).

(This PR is simply copied from the PR linked above, so see that for rationale.)

Closes #72165.

Before, Firefox on Linux:

image

After, Firefox on Linux (`prefers-color-scheme` set to `dark` by setting `ui.systemUsesDarkTheme` to a number value of `1` in `about:config`):

image

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
I am in a dark theme and yet the favicon are all visible. I'm not sure how reliable this is overall...

**Comment by Cldfire** [comment]
> I am in a dark theme and yet the favicon are all visible. I'm not sure how reliable this is overall...

Yeah, `prefers-color-scheme` isn't super widespread yet. This is still an improvement for many people over the way it was before (no attempt to handle dark themes), though.

We could just stick a white outline around the favicon like what we did for the logo, but I think there's some value in doing things the same way other Rust web properties have chosen to do it.

**Comment by GuillaumeGomez** [comment]
> We could just stick a white outline around the favicon like what we did for the logo, but I think there's some value in doing things the same way other Rust web properties have chosen to do it.

I'd prefer that if you don't mind. Let's try to keep only one strategy overall.

**Comment by Cldfire** [comment]
Well, I tried the white outline approach:

image

But it just doesn't look good for the favicon:



There aren't enough pixels to work with to make the outline work.

I also tried just using the fallback PNG favicons from [the PR](https://github.com/rust-lang/www.rust-lang.org/pull/1185):

image

But I really don't like the look of that either. It just feels... wrong.

I would like to stick with the SVG favicon because:

1. It looks great
2. It is consistent with the other Rust websites, so if the user gets the right favicon color everywhere else they'll get it in Rustdoc too

Note that [mdbook also uses this approach](https://github.com/rust-lang/mdBook/pull/1230).

**Comment by Cldfire** [comment]
Also note that this is the same approach GitHub uses for their favicon. `ui.systemUsesDarkTheme` set to `0`:

image
 
Set to `1`:

image

I've also seen this approach widely recommended online. It may not be perfectly supported now but it seems to be the future.

**Comment by crlf0710** [comment]
Ping from triage, what's the current status on this? Also i saw there's a merge commit in this PR, which is unacceptable. @Cldfire needs to rebase the commits before it can be accepted.

**Comment by Cldfire** [comment]
> Ping from triage, what's the current status on this?

Waiting on approval / rejection from the rustdoc team.

> Also i saw there's a merge commit in this PR, which is unacceptable. @Cldfire needs to rebase the commits before it can be accepted.

Rebased :slightly_smiling_face:

**Comment by jyn514** [comment]
ping @GuillaumeGomez (@Manishearth might have opinions too since they were a big part of the decision for https://github.com/rust-lang/rust/pull/75249)

File: src/librustdoc/html/layout.rs
Comment (GuillaumeGomez) [inline src/librustdoc/html/layout.rs:163]: Shouldn't the order be the "icon" first and the "alternate" then?

**Comment by GuillaumeGomez** [comment]
Just a nit/question but otherwise looks good to me.

File: src/librustdoc/html/layout.rs
Comment (Cldfire) [reply src/librustdoc/html/layout.rs:163]: Hmm, I'm not sure? It was in this order in the original PR, and looking at [this page](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link) it would appear that `alternate` comes first:

```
<link href="default.css" rel="stylesheet" title="Default Style">
<link href="fancy.css" rel="alternate stylesheet" title="Fancy">
<link href="basic.css" rel="alternate stylesheet" title="Basic">
```

But I'm not sure if there's a spec somewhere for it or if it even matters.

Comment (GuillaumeGomez) [reply src/librustdoc/html/layout.rs:163]: From the code you showed, non-alternate comes first. ;)

Comment (Cldfire) [reply src/librustdoc/html/layout.rs:163]: Oh sorry, my bad! I thought you were referring to the ordering of the entries in the `rel` string :joy:

I'll move the `icon` only link to the top later today 👍

Comment (GuillaumeGomez) [reply src/librustdoc/html/layout.rs:163]: Thanks! ;)

**Comment by Manishearth** [comment]
No opinions, as long as this is just for the Rust logo I'm fine with it

File: src/librustdoc/html/layout.rs
Comment (Cldfire) [reply src/librustdoc/html/layout.rs]: Sorry, I got busy yesterday. Fixed :)

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**
